metadata_backend/api/operators/file.py

```diff
@@ -193,18 +202,61 @@ async def read_project_files(self, project_id: str) -> List[Dict]:
             LOG.exception(reason)
             raise web.HTTPInternalServerError(reason=reason) from error
 
+    async def check_submission_files_ready(self, submission_id: str) -> None:
+        """Check all files in a submission are marked as ready.
+
+        Files marked as ready in a submission, means an metadata object has been
+        attached to the file.
+
+        :param submission_id: Submission ID to get associated files status
+        :raises: HTTPInternalServerError if db operation failed because of connection
+        or other db issue
+        """
+        aggregate_query = [
+            {"$match": {"submissionId": submission_id}},
+            {"$unwind": "$files"},
+            # check the status is not in failed or added
+            # failed can occour when an file ingestion/verification/mapping fails
+            {"$match": {"files.status": {"$in": ["added", "failed"]}}},
+            {
+                "$project": {
+                    "_id": 0,
+                    "accessionId": "$files.accessionId",
+                    "version": "$files.version",
+                    "status": "$files.status",
+                }
+            },
+        ]
+        try:
+            problematic_files = await self.db_service.do_aggregate("submission", aggregate_query)
+            if len(problematic_files) > 0:
+                reason = (
+                    f"There are a problematic files: {','.join([i['accessionId'] for i in problematic_files])} "
+                    f"in the submission with id: {submission_id}"
+                )
+                LOG.error(reason)
+                raise web.HTTPBadRequest(reason=reason)
```

Comment:
not sure this should be bad status, maybe 422, or smth else, any suggestions?

Reply:
I think that 400 is good.



_The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing)._

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
Although the examples are quite technical, it seems to me that the other return codes are meant for more specific cases.

I'd consider "forgetting" to either remove the files or attach them to an object as a client error ^^

Reply:
yes, the plan is to attach them to the error object but for that we might need #641

Reply:
fixed with #642  and https://github.com/CSCfi/metadata-submitter/pull/633/commits/90fef6fa0a2d199967eaea403949fabac1f419d6